### PR TITLE
[10.0][ADD] website_blog_category: New module

### PR DESCRIPTION
**Pull request by lasley**
Website Blog - Categories (#328)
=========================

This module provides the ability to categorize blog posts.

Usage
=====

To use this module:
-------------------

* Go to `Website Admin` => `Blog Categories` to create blog categories.
* Enable category browsing in Blog website customization options
* Category browser is displayed in blog sidebar

### CODE REVIEW

**Comment by lasley** [comment]
@tedsalmon please review

File: .travis.yml
Comment (lasley) [inline .travis.yml]: Switch back after OCA/maintainer-quality-tools#437 merge

**Review by tedsalmon — APPROVED** [review]
LGTM! 👍 
Any plans to up the coverage though?
Coveralls is showing an average of 71% coverage
```
Name                                              Stmts   Miss  Cover   Missing
-------------------------------------------------------------------------------
website_blog_category/controllers/main.py            14      7    50%   21-33
website_blog_category/models/blog_blog.py             4      0   100%
website_blog_category/models/blog_category.py        31      0   100%
website_blog_category/models/blog_post.py            12      2    83%   24-25
website_field_autocomplete/controllers/main.py       16      9    44%   22-26, 38-43
website_snippet_data_slider/controllers/main.py      18     10    44%   28-39
-------------------------------------------------------------------------------
TOTAL                                                95     28    71%
```

**Comment by lasley** [comment]
@tedsalmon - interesting, those lines are all covered with my tour. Looks like there was some sort of [PhantomJS timeout](https://travis-ci.org/OCA/website/jobs/213946845#L938) though, which likely affected the coverage.

**Review by yajo — APPROVED** [review]
code review ok

**Comment by yajo** [comment]
Although the code is OK, I wonder why do we need this at all? Blog already has tags...

**Comment by lasley** [comment]
Tags and categories are mutually exclusive methods of organization. Categories are typically more of a hierarchical layout vs tags are more about the amount of content classified under the specific tag (tag cloud). 

This is required in order for me to migrate my Wordpress blog, which contains both concepts, into Odoo.

**Comment by yajo** [comment]
Hmm I know. Wordpress is weird on that IMHO; Blogger for instance has only tags.

I would not complicate this stuff and just use tags, but in any case I'm not blocking just for that 😉

**Comment by lasley** [comment]
In order to use the Odoo tags, I would still need to create a hierarchal structure which does not currently exist. It seemed to me that adding this structure on top of a pre-existing flat structure would not play nicely with information that is already in the system. 

It should also be noted that the concept of labels in Blogger are actually more akin to categories, due to their use as menu items & the ability for sub-labels. 

Honestly I'd be fine outright removing Odoo tags in order to keep things simple, but yeah that seems like a pretty profound change for a module. Same with making the tags function like categories.

Thoughts?

**Comment by yajo** [comment]
Hmmm yeah, given that use case this one seems best option 😉

**Pull request merged**
